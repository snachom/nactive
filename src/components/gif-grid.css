.search-name, .loader {
  text-align: center;
}

.card-grid {
  display: flex;
  flex-flow: row wrap;
  /* flex-flow: column wrap; */
  width: 100%;
  justify-content: center;

}