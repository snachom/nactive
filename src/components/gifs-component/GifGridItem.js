import React from 'react';
import PropTypes from 'prop-types';

import './gif-grid-item.css'

export const GifGridItem = ( {id, title, url} ) => {

  return (
      <>
        <div className="gif-card animate__animated animate__fadeIn">
          <img src={url} alt={title} />
          <p> {title}</p>
        </div>
      </>
    );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}