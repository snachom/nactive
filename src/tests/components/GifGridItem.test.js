import {shallow} from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

  const title = 'Un título'
  const url = 'https://localhost/algo.jpg'
  const wrapper = shallow( <GifGridItem title={title} url={url}/>)

  test('debe mostrar el componente correctamente', () => {

    expect(wrapper).toMatchSnapshot();

  })

  test('debe tener un párrafo con el título', () => {

    const p = wrapper.find('p')
    expect(p.text().trim()).toBe( title )

  })

  test('debe tener la imagen igual al url y alt de los props', () => {

    const img = wrapper.find('img')
    // console.log( img.props('src') )
    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)

  })

  test('deber tener animate__fadeIn', () => {

    const div = wrapper.find('div')
    const className = div.prop('className')
    // console.log(div.props('className'))

    expect(className.includes('animate__fadeIn')).toBe(true)

  })
});
