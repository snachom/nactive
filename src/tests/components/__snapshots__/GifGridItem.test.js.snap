// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<Fragment>
  <div
    className="card animate__animated animate__fadeIn"
  >
    <img
      alt="Un título"
      src="https://localhost/algo.jpg"
    />
    <p>
       
      Un título
    </p>
  </div>
</Fragment>
`;
